.github/workflows/codeql-analysis.yml

```diff
@@ -0,0 +1,39 @@
+name: "CodeQL"
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+    branches: [main]
+  schedule:
+    - cron: '0 22 * * 3'
+
+jobs:
+  analyze:
+    name: Analyze
+    runs-on: ubuntu-latest
+
+    strategy:
+      fail-fast: false
+      matrix:
+        language: ['python']
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 2
+      
+      - run: git checkout HEAD^2
```

Comment:
@iMartzen this seems odd, can you check?

Reply:
I copy + paste the template from `react-date-picker` repo 😅, so I needed to find out why this was used. 

> The use of `git checkout HEAD^2` in the `codeql.yaml` is meant to ensure that the workflow checks out the correct commit when the run is triggered by a pull request event. In GitHub Actions, when a pull request event triggers a workflow, the default behavior is to check out the merge commit created by GitHub. This merge commit represents the state of the branch after merging the pull request into the base branch. While this is useful for some checks, there are cases where you might want to analyze the actual head commit of the pull request instead.
> 
> Here's a breakdown of what `HEAD^2` does:
> 
> - `HEAD` refers to the current commit checked out by GitHub Actions, which is the merge commit in the context of a pull request.
> - `HEAD^2` refers to the second parent of the merge commit. In a typical merge commit, the first parent (`HEAD^`) is the commit on the base branch (e.g., `main`), and the second parent (`HEAD^2`) is the commit from the head branch (i.e., the commit from the pull request branch).
> 
> By using `git checkout HEAD^2`, you ensure that the workflow checks out the head commit of the pull request, rather than the merge commit. This can be useful if you need to run analysis or tests specifically on the changes introduced by the pull request without considering the merge with the base branch.
> 
> If your goal is to analyze the changes introduced by the pull request itself, using `HEAD^2` is the correct approach. However, if you want to analyze the state of the branch after the pull request would be merged, you should stick with the default behavior.